src/private/m2tUpdater.m

```diff
@@ -58,180 +171,165 @@
-                      movefile(unzippedFiles{ii}, unzippedFilesTarget{ii})
-                  end
-                  % Add topZipFolder to current folder structure
-                  currentFolderFiles = [currentFolderFiles; fullfile(targetPath, topZipFolder{1})];
-              end
-              
-              % Cleanup
-              newFolderStructure = [getFolders(unzippedFilesTarget);  unzippedFilesTarget];
-              deleteFolderFiles  = setdiff(currentFolderFiles, newFolderStructure);
-              for ii = 1:numel(deleteFolderFiles)
-                  x = deleteFolderFiles{ii};
-                  if exist(x, 'file')
-                      delete(x);
-                  elseif exist(x, 'dir')
-                      rmdir(x,'s');
-                  end
-              end
-              
-              upgradeSuccess = true; %~isempty(unzippedFiles);
-              userInfo(verbose, 'UPDATED: the current conversion will be terminated. Please, re-run it.');
-          catch
-              userInfo(verbose, ['FAILED: continuing with the' name ' conversion.']);
-          end
-      end
-      userInfo(verbose, '');
-  end
 end
-% =========================================================================
-function isBelow = isVersionBelow(env, versionA, versionB)
-  % Checks if version string or vector versionA is smaller than
-  % version string or vector versionB.
-
-  vA = versionArray(env, versionA);
-  vB = versionArray(env, versionB);
-
-  isBelow = false;
-  for i = 1:min(length(vA), length(vB))
-    if vA(i) > vB(i)
-      isBelow = false;
-      break;
-    elseif vA(i) < vB(i)
-      isBelow = true;
-      break
+% ==============================================================================
+function askToShowChangelog(currentVersion)
+% Asks whether the user wants to see the changelog and then shows it.
+    reply = input(' *** Would you like to see the changelog? y/n [y]:' ,'s');
+    shouldShow =  isempty(reply) || ~strcmpi(reply(1),'n') ;
+    if shouldShow
+        fprintf(1, '\n%s\n', changelogUntilVersion(currentVersion));
     end
-  end
+end
+% ==============================================================================
+function changelog = changelogUntilVersion(currentVersion)
+% This function retrieves the chunk of the changelog until the current version.
+    URL = 'https://github.com/matlab2tikz/matlab2tikz/raw/master/CHANGELOG.md';
+    changelog = urlread(URL);
+    currentVersion = versionString(currentVersion);
+
+    % Header is "# YYYY-MM-DD Version major.minor.patch [Manager](email)"
+    % Just match for the part until the version number. Here, we're actually
+    % matching a tiny bit too broad due to the periods in the version number
+    % but the outcome should be the same if we keep the changelog format
+    % identical.
+    pattern = ['\#\s*[\d-]+\s*Version\s*' currentVersion];
+    idxVersion = regexpi(changelog, pattern);
+    if ~isempty(idxVersion)
```

Comment:
I agree. When the user is on a version without a proper changelog entry, the whole changelog is shown. But in general, that situation should not happen (outside of manual testing).

I slightly prefer the current approach, where we just show the whole thing. Most importantly because it is the least complicated way (hence, less chance for bugs) and because the feature is just a convenience, not an actual must-have feature.